File: application-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/config/__pluginName__ApplicationConfig.java
package ${package}.config;

import se.curity.identityserver.sdk.config.Configuration;
import se.curity.identityserver.sdk.config.annotation.Description;

import se.curity.identityserver.sdk.service.SessionManager;

public interface ${pluginName}ApplicationConfig extends Configuration
{
    @Description("A parameter used for demo purposes")
    Integer getDemoParameter();

    SessionManager getSessionManager();
}

File: kotlin-application-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/config/__pluginName__ApplicationConfig.kt
package ${package}.config;

import se.curity.identityserver.sdk.config.Configuration;
import se.curity.identityserver.sdk.config.annotation.Description;

import se.curity.identityserver.sdk.service.SessionManager

interface ${pluginName}ApplicationConfig: Configuration
{
    @Description("A parameter used for demo purposes")
    fun getDemoParameter(): Int;

    fun getSessionManager(): SessionManager;
}
